File: tp1-codigos/SomaRec.c
#include <stdio.h>
#include <stdlib.h>

//soma recursivamente, se tiver apenas um digito é o caso base, nao precisa de soma, se tiver mais de um digito, chama a funçao removendo o ultimo digito + ultimo digito
int soma(int n){
	if(n <= 9) return n;
	else return soma(n/10) + n%10;
}

int main(){
	int num;

	while(scanf("%d", &num) != EOF){
		printf("%d\n", soma(num));
	}

	return 0;
}

File: tp1-codigos/SomaRec.java
import java.util.Scanner;

class SomaRec{
	
//funcao para somar os digitos, quando se for apenas um digito retorna ele mesmo, caso contrario, a chamada da funçao retira o ultimo digito com n/10 e soma com o digito apos a chamada
	public static int Soma(int n){
		if(n <= 9) return n;
		else return Soma(n/10) + n%10;
	}

	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);

		while(sc.hasNextInt()){
			int num = sc.nextInt();
			System.out.println(Soma(num));
		}
		sc.close();
	}
}
